addons/mrp/models/mrp_workorder.py

```diff
@@ -344,8 +344,6 @@ def _float_duration_to_second(duration):
             delta_duration = new_order_duration - old_order_duration
 
             if delta_duration > 0:
-                if order.state not in ('progress', 'done'):
-                    order.state = 'progress'
```

Comment:
Why do you remove that? Seems unrelated to the bug. :thinking: 
Also, since it will add a time sheet for the added time, there's little difference between doing that *or* starting then pausing a workorder. Yet in the first case the wo would still be `ready` and in the other case it would be `progress`.